### PR TITLE
[Merged by Bors] - fix: shallow partial equals (VF-3887)

### PR DESCRIPTION
**Pull request by DecathectZero**
**Fixes or implements VF-3887**

### Brief description. What is this change?
We need this to not trigger the `onChange` function in the FE if people don't actually change anything.

### CODE REVIEW

File: packages/common/src/utils/object/common.ts [hunk 1]
```diff
@@ -92,3 +92,7 @@ export const filterEntries = pickBy;
 
 export const mapValue = <T, R>(obj: Record<string | number | symbol, T>, callback: (value: T) => R) =>
   Object.fromEntries(Object.entries(obj).map(([key, value]) => [key, callback(value)]));
+
+export const shallowPartialEquals = (obj: unknown, partial: unknown) => {
+  return isObject(obj) && isObject(partial) && Object.keys(partial).every((key) => partial[key] === obj[key]);
```
Comment (jonahsnider) [inline packages/common/src/utils/object/common.ts:97]:
```suggestion
  return isObject(obj) && isObject(partial) && Object.entries(partial).every(([key, partialValue]) => partialValue === obj[key]);
```

File: packages/common/src/utils/object/common.ts [hunk 2]
```diff
@@ -92,3 +92,7 @@ export const filterEntries = pickBy;
 
 export const mapValue = <T, R>(obj: Record<string | number | symbol, T>, callback: (value: T) => R) =>
   Object.fromEntries(Object.entries(obj).map(([key, value]) => [key, callback(value)]));
+
+export const shallowPartialEquals = (obj: unknown, partial: unknown) => {
```
Comment (effervescentia) [inline packages/common/src/utils/object/common.ts:96]: I don't know if we should be just operating on unknown values, this kind of makes this function less useful, if we add types we can make it more clear the relationship between `obj` and `partial`

```suggestion
export const shallowPartialEquals = <T extends object>(obj: T, partial: Partial<T>) => {
```

File: packages/common/src/utils/object/common.ts (hunk 1 above)
Comment (DecathectZero) [reply packages/common/src/utils/object/common.ts:97]: I'm very surprised by this lol.
Screen Shot 2022-06-22 at 1 37 00 PM

**Comment by effervescentia** [comment]
since we're adding a standard `shallowPartialEquals` I suggest we should also export `shallowequal` as something like `shallowEquals` from the same `object.ts` file so we have both available which helps underline the differences

**Review by effervescentia — APPROVED** [review]

**Pull request closed without merging**